apps/server/src/controllers/userControllers/authUsers.ts

```diff
@@ -291,3 +299,69 @@ export const forgotPassword = async (req: Request, res: Response) => {
     })
   }
 }
+
+export const resetPassword = async (req: Request, res: Response) => {
+  try {
+    const { verify: longString } = req.query
+
+    // Check for reset-token instance using query-string(id)
+    const resetTokenInstance = await UserResetPasswordToken.findOne({
+      where: { id: longString as string, used: false },
+    })
+
+    if (!resetTokenInstance) {
+      return res.status(HTTP_STATUS_CODE.UNAUTHORIZED).json({
+        message: 'Invalid or already used verification token.',
+      })
+    }
+
+    const { id, token } = resetTokenInstance
+
+    // Verify the token with the jwt secret & query-string(id)
+    const decodedToken = await Jwt.verify(token, {
+      _secret: Env.JWT_SECRET + id,
+    })
+
+    if (decodedToken) {
+      const validResetBody = resetPasswordSchema.strict().safeParse(req.body)
+
+      if (!validResetBody.success) {
+        return res.status(HTTP_STATUS_CODE.BAD_REQUEST).json({
+          message: 'Invalid request body.',
+          errors: validResetBody.error,
+        })
+      }
+
+      const { newPassword, confirmPassword } = validResetBody.data
+
+      if (newPassword !== confirmPassword) {
+        return res.status(HTTP_STATUS_CODE.CONFLICT).json({
+          message: 'Password mismatch',
+        })
+      }
```

Comment:
conflict! there's no confirm password

Reply:
I thought I saw a confirm password field in the figma??

Reply:
that's a frontend thing